test/e2e/framework/service/jig.go

```diff
@@ -1051,3 +958,121 @@ func (j *TestJig) CreateSCTPServiceWithPort(tweak func(svc *v1.Service), port in
+			continue
+		}
+		isNodeInternalIPV4 := net.ParseIP(internalAddr).To4() != nil
+		// Check service reachability on the node internalIP which is same family
+		// as clusterIP
+		if isClusterIPV4 != isNodeInternalIPV4 {
+			framework.Logf("skipping testEndpointReachability() for internal adddress %s as it does not match clusterIP (%s) family", internalAddr, clusterIP)
+			continue
+		}
+
+		err := j.testEndpointReachability(internalAddr, sp.NodePort, sp.Protocol, pod)
+		if err != nil {
+			return err
+		}
+	}
+	for _, externalAddr := range externalAddrs {
+		isNodeExternalIPV4 := net.ParseIP(externalAddr).To4() != nil
+		if isClusterIPV4 != isNodeExternalIPV4 {
+			framework.Logf("skipping testEndpointReachability() for external adddress %s as it does not match clusterIP (%s) family", externalAddr, clusterIP)
+			continue
+		}
+		err := j.testEndpointReachability(externalAddr, sp.NodePort, sp.Protocol, pod)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// isInvalidOrLocalhostAddress returns `true` if the provided `ip` is either not
+// parsable or the loopback address. Otherwise it will return `false`.
+func isInvalidOrLocalhostAddress(ip string) bool {
+	parsedIP := net.ParseIP(ip)
+	if parsedIP == nil || parsedIP.IsLoopback() {
+		return true
+	}
+	return false
+}
+
+// testEndpointReachability tests reachability to endpoints (i.e. IP, ServiceName) and ports.  Note that the execPod
+// must have nc installed.
+func (j *TestJig) testEndpointReachability(endpoint string, port int32, protocol v1.Protocol, execPod *v1.Pod) error {
+	// 1) Wait for the pod to come up before we test endpoints
+	e2epod.WaitForPodCondition(j.Client, execPod.ObjectMeta.Namespace, execPod.ObjectMeta.Name, "", ServiceReachabilityShortPollTimeout,
+		func(pod *v1.Pod) (bool, error) {
+			framework.Logf("Waiting for pod %v to be running, current state %v", execPod.ObjectMeta.Name, pod.Status.Phase)
+			return pod.Status.Phase == v1.PodRunning, nil
+		})
+
+	// 2) make sure the exec pod is valid before attempting to poll nc, otherwise when misusing this function
+	// alot of spam can get created.
+	err := wait.PollImmediate(1*time.Second, ServiceReachabilityShortPollTimeout, func() (bool, error) {
+		if _, err := framework.RunHostCmd(execPod.Namespace, execPod.Name, "which nc"); err != nil {
+			framework.Logf("verifying execPod can run 'nc': %v", err)
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		fmt.Errorf("nc must be installed on the target pod for testEndpointReachability to work properly")
+	}
+
+	// 3) Finally take care of the polling
+	ep := net.JoinHostPort(endpoint, strconv.Itoa(int(port)))
+	cmd := ""
+	switch protocol {
+	case v1.ProtocolTCP:
+		cmd = fmt.Sprintf("nc -zv -t -w 2 %s %v", endpoint, port)
+	case v1.ProtocolUDP:
+		cmd = fmt.Sprintf("nc -zv -u -w 2 %s %v", endpoint, port)
+	case v1.ProtocolSCTP:
+		cmd = fmt.Sprintf("nc -zv --sctp -w 2 %s %v", endpoint, port)
```

Comment:
what netcat version is this? the one provided in the agnhost  image used by most of the test does not have this option
```
 nc -h
OpenBSD netcat (Debian patchlevel 1.130)
```

Reply:
This function takes the pod as *input* so .  That said you are right, many pods might not support the `--sctp` option.  However *some pods* will (specifically, we need to support this for sure in #91592, so we'll make sure of that)

Reply:
Looking more

- it looks like `netcat-openbsd` doesnt support sctp
- `netcat` the regular package *does* support sctp

But again, this function's ability to use sctp is going to be container dependent, regardless, and so it seems like its reasonable to say ' if the pod has an the current version of  nc, this will work, if it has an older version, it might not work'.

Reply:
Trying this out in a related issue #96482 . Once you install the nmap-netcat, does it symlink `nc` to `ncat` ?

Reply:
Yup it does (correct me if I'm wrong here been a few months since I put this patch up)

Reply:
I don't think that it did, at least when I tried the test with your change to the Docker image, it just seemed to pickup the nc that is installed in BusyBox. 

I added the changes and tried to use the new ncat explicitly in the following branch that you can try out https://github.com/cmluciano/kubernetes/tree/cml/testnchelp

`export KUBE_TEST_REPO_LIS=$GOPATH/src/k8s.io/kubernetes/repo_list.yaml`
```bazel build //test/e2e:e2e.test && \
 cd src/k8s.io/kubernetes/bazel-out/k8-fastbuild-ST-77d62fd15d15fc7ab69058da1afba983b16c05cd7d7cd724cce6b37445e7a9d1/bin/test/e2e/e2e.test_ \
&& ./e2e.test -context kind-kind -ginkgo.focus="RunUdpTest" -ginkgo.trace```

When I try that out, my output shows nc pointing to the busybox version and it is symlinked to the normal `/usr/bin/nc`.